oteljava/all/src/main/scala/org/typelevel/otel4s/oteljava/OtelJava.scala

```diff
@@ -97,6 +103,30 @@ object OtelJava {
       )
       .evalMap(forAsync[F])
 
+  /** Creates a [[cats.effect.Resource `Resource`]] of the automatic
+    * configuration of a Java `OpenTelemetrySdk` instance.
+    *
+    * If you rely on
+    * [[https://opentelemetry.io/docs/instrumentation/java/automatic/ automatic instrumentation via Java agent]],
+    * you MUST NOT use this method and MUST use [[global]] instead.
+    *
+    * @param customize
+    *   A function for customizing the auto-configured SDK builder. This
+    *   function MUST NOT call `setResultAsGlobal`.
+    * @return
+    *   An [[org.typelevel.otel4s.Otel4s]] resource.
+    */
+  def autoConfigured[F[_]: LiftIO: Async](
+      customize: AutoConfigOtelSdkBuilder => AutoConfigOtelSdkBuilder = identity
+  ): Resource[F, OtelJava[F]] =
+    resource {
+      Sync[F].delay {
+        customize(AutoConfigOtelSdk.builder())
+          .build()
```

Comment:
Since we manually manage the lifecycle (via Resource), should we disable the JVM shutdown hook? 

```suggestion
        customize(AutoConfigOtelSdk.builder())
          .disableShutdownHook()
          .build()
```

Reply:
I didn't realise that—good catch!

Reply:
hmm... my only concern is that if someone interrupts the JVM (`[ctrl]`+`c` or equivalent) it won't get shut down. and it looks like it has no issue with calling shutdown twice; it just logs a message and returns success. and according to the docs:

> Returns: a `CompletableResultCode` which completes when all providers are shutdown

it doesn't say that it can only be called once, or that it throws when called twice, so would it be safer to just let it happn anyway?

Reply:
> hmm... my only concern is that if someone interrupts the JVM (`[ctrl]`+`c` or equivalent)

If you are using `IOApp` and `Resource` properly, then this will never happen. `IOApp` always runs finalizers when it gets a ctrl+c.